### PR TITLE
adds two notebooks to test clearing outputs

### PR DESCRIPTION
**Pull request by bryanbritten**
# Pull Request

## Description
Adds two notebooks that I have been working on to allow other team members to build upon and utilize this work.

## Related Issues
N/A

## Additional Notes
The primary objective of this PR is to test that the `.git/config` option I implemented automatically strips output cells and metadata from the notebook, reducing the chance for PHI/PII to make it into the repo, as well as standardizing what parts of a notebook are version controlled.

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: dedupe/notebooks/lac_ecr_eda.ipynb [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+    "    def get_patient_name(self):\n",
+    "        given = self.root.find('.//default:patient/default:name/default:given', self.ns).text\n",
+    "        family = self.root.find('.//default:patient/default:name/default:family', self.ns).text\n",
+    "        return f\"{given} {family}\"\n",
+    "    \n",
+    "    def get_patient_dob(self):\n",
+    "        return self.root.find('.//default:patient/default:birthTime', self.ns).attrib['value']\n",
+    "    \n",
+    "    def get_send_date(self):\n",
+    "        time = self.root.find('.//default:effectiveTime', self.ns)\n",
```
Comment (alhayward) [inline dedupe/notebooks/lac_ecr_eda.ipynb:114]: Can we document somewhere our assumptions about data field definitions/mappings? (E.g., what is send date, which XML field is send date, and why)

File: dedupe/notebooks/lac_ecr_eda.ipynb [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+    "        if time is not None:\n",
```
Comment (alhayward) [inline dedupe/notebooks/lac_ecr_eda.ipynb:115]: How are we interpreting when an eCR field is `None`? Are we assuming the data is unavailable, potential data entry error, something else?

Understanding nullity in the data will be important (e.g., how many eCRs have null DOB?), so maybe if the value is `None`, we can still return `None` to capture this.

File: dedupe/notebooks/lac_ecr_eda.ipynb [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+    "    def __init__(self, path):\n",
+    "        super().__init__(path)\n",
+    "    \n",
+    "    def get_patient_id(self):\n",
+    "        return self.root.find('.//default:patient/default:id', self.ns).attrib['extension']\n",
+    "    \n",
+    "    def get_patient_name(self):\n",
+    "        given = self.root.find('.//default:patient/default:name/default:given', self.ns).text\n",
+    "        family = self.root.find('.//default:patient/default:name/default:family', self.ns).text\n",
+    "        return f\"{given} {family}\"\n",
```
Comment (alhayward) [inline dedupe/notebooks/lac_ecr_eda.ipynb:108]: What if these values are `None`? Same for patient ID, DOB

File: dedupe/notebooks/lac_ecr_eda.ipynb [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+   "source": [
+    "section_name_counts = {}\n",
+    "\n",
+    "for dir in all_dirs:\n",
+    "    # read in the CDA_eICR.xml file, get the sections, and print the section titles\n",
+    "    file_path = Path(\"./../data/LAC_DATA/\" + dir + \"/CDA_eICR.xml\")\n",
+    "    ecr = eICR(file_path)\n",
+    "    sections = ecr.get_sections()\n",
+    "    for section in sections:\n",
+    "        section_name = section.find('.//default:title', ecr.ns).text\n",
```
Comment (alhayward) [inline dedupe/notebooks/lac_ecr_eda.ipynb:155]: I wonder if LOINC/SNOWMED codes are more standard than section titles across eCRs. I'll look into this!

**Review by alhayward — COMMENTED** [review]
Thanks for putting this together! Pulling and executing the notebooks worked for me 👍🏼

While the `notebooks/` dir works to share this code for now, if we are building upon this code I think modularizing some of it would allow us to more easily review and contribute to the classes/functions that may be helpful for working with eCRs outside the analysis of this notebook. Here are some ideas:

`ecr/processing.py`
* `extract_zip_files()`
* `extract_cell_data()`
* `parse_section_content()`
* `clean()`

`ecr/eCR.py`
* `eCR()`
* `RR()`
* `eICR()`

Happy to work on this if the team think it's the right direction! Also open to pushback - maybe this code should only live in the context of the LAC eCR data analysis.

File: dedupe/notebooks/lac_ecr_eda.ipynb (hunk 4 above)
Comment (bryanbritten) [reply dedupe/notebooks/lac_ecr_eda.ipynb:155]: I believe this is definitely the case and is likely the better way to extract sections. Adding that analysis to this notebook would be very beneficial!

File: dedupe/notebooks/lac_ecr_eda.ipynb (hunk 1 above)
Comment (bryanbritten) [reply dedupe/notebooks/lac_ecr_eda.ipynb:114]: I think this is definitely worth doing!
